File: app/src/main/java/com/scatl/uestcbbs/module/account/adapter/AccountSwitchAdapter.java
package com.scatl.uestcbbs.module.account.adapter;

import android.view.View;
import android.widget.RadioButton;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.AccountBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;
import com.scatl.uestcbbs.util.SharePrefUtil;

/**
 * author: sca_tl
 * date: 2021/9/20 10:12
 * description:
 */
public class AccountSwitchAdapter extends BaseQuickAdapter<AccountBean, BaseViewHolder> {

    private int currentSelectUid;

    public AccountSwitchAdapter(int layoutResId) {
        super(layoutResId);
    }

    public void setCurrentSelectUid(int currentSelectUid) {
        this.currentSelectUid = currentSelectUid;
    }

    @Override
    protected void convert(BaseViewHolder helper, AccountBean item) {
        helper.setText(R.id.item_account_switch_name, item.userName)
                .setText(R.id.item_account_switch_account_status, SharePrefUtil.isSuperLogin(mContext, item.userName) ? "已高级授权" : "未高级授权");
        RadioButton radioButton = helper.getView(R.id.item_account_switch_radiobtn);
        radioButton.setChecked(currentSelectUid == item.uid);
        GlideLoader4Common.simpleLoad(mContext, item.avatar, helper.getView(R.id.item_account_switch_avatar));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/account/adapter/AccountSwitchAdapter.kt
package com.scatl.uestcbbs.module.account.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemAccountSwitchBinding
import com.scatl.uestcbbs.entity.AccountBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.SharePrefUtil
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/9 15:15
 */
class AccountSwitchAdapter: PreloadAdapter<AccountBean, ItemAccountSwitchBinding>() {

    var currentLoginUid: Int = 0

    override fun getViewBinding(parent: ViewGroup): ItemAccountSwitchBinding {
        return ItemAccountSwitchBinding.inflate(LayoutInflater.from(parent.context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemAccountSwitchBinding>, position: Int, item: AccountBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.name.text = item.userName
        holder.binding.cookiesStatus.text = if (SharePrefUtil.isSuperLogin(context, item.userName)
            && !SharePrefUtil.getCookies(context, item.userName).isNullOrEmpty()) {
            "已获取Cookies"
        } else {
            "未获取Cookies，部分功能受限"
        }
        holder.binding.radioBtn.isChecked = currentLoginUid == item.uid
        holder.binding.avatar.load(item.avatar)
    }
}